pkg/oonimkall/taskrunner_test.go

```diff
@@ -546,62 +491,25 @@ func TestTaskRunnerRun(t *testing.T) {
 	})
 
-	t.Run("with success and InputStrictlyRequired", func(t *testing.T) {
+	t.Run("with success and more than one entry", func(t *testing.T) {
+		inputs := []string{"a", "b", "c", "d"}
 		runner, emitter := newRunnerForTesting()
-		runner.settings.Inputs = []string{"a", "b", "c", "d"}
+		runner.settings.Inputs = inputs // this is basically ignored because we override MockLoad
 		fake := fakeSuccessfulDeps()
-		fake.Builder.MockInputPolicy = func() model.InputPolicy {
-			return model.InputStrictlyRequired
-		}
-		runner.newSession = fake.NewSession
-		events := runAndCollect(runner, emitter)
-		reduced := reduceEventsKeysIgnoreLog(t, events)
-		expect := []eventKeyCount{
-			{Key: eventTypeStatusQueued, Count: 1},
-			{Key: eventTypeStatusStarted, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 3},
-			{Key: eventTypeStatusGeoIPLookup, Count: 1},
-			{Key: eventTypeStatusResolverLookup, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 1},
-			{Key: eventTypeStatusReportCreate, Count: 1},
-			//
-			{Key: eventTypeStatusMeasurementStart, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 1},
-			{Key: eventTypeMeasurement, Count: 1},
-			{Key: eventTypeStatusMeasurementSubmission, Count: 1},
-			{Key: eventTypeStatusMeasurementDone, Count: 1},
-			//
-			{Key: eventTypeStatusMeasurementStart, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 1},
-			{Key: eventTypeMeasurement, Count: 1},
-			{Key: eventTypeStatusMeasurementSubmission, Count: 1},
-			{Key: eventTypeStatusMeasurementDone, Count: 1},
-			//
-			{Key: eventTypeStatusMeasurementStart, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 1},
-			{Key: eventTypeMeasurement, Count: 1},
-			{Key: eventTypeStatusMeasurementSubmission, Count: 1},
-			{Key: eventTypeStatusMeasurementDone, Count: 1},
-			//
-			{Key: eventTypeStatusMeasurementStart, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 1},
-			{Key: eventTypeMeasurement, Count: 1},
-			{Key: eventTypeStatusMeasurementSubmission, Count: 1},
-			{Key: eventTypeStatusMeasurementDone, Count: 1},
-			//
-			{Key: eventTypeStatusEnd, Count: 1},
-		}
-		assertReducedEventsLike(t, expect, reduced)
-	})
-
-	t.Run("with success and InputOptional and input", func(t *testing.T) {
-		runner, emitter := newRunnerForTesting()
-		runner.settings.Inputs = []string{"a", "b", "c", "d"}
-		fake := fakeSuccessfulDeps()
-		fake.Builder.MockInputPolicy = func() model.InputPolicy {
-			return model.InputOptional
+		fake.Builder.MockNewTargetLoader = func(config *model.ExperimentTargetLoaderConfig) model.ExperimentTargetLoader {
+			return &mocks.ExperimentTargetLoader{
+				MockLoad: func(ctx context.Context) (targets []model.ExperimentTarget, err error) {
+					// We need to mimic wht would happen when settings.Inputs is explicitly provided
+					for _, input := range inputs {
+						targets = append(targets, model.NewOOAPIURLInfoWithDefaultCategoryAndCountry(input))
+					}
+					return
+				},
+			}
```

Comment:
Like before, we can remove test cases because we treat experiments uniformly since we use a target loader.